File: ts-rosalind/src/fib.ts

export function fib(n: number, k: number): BigInt {
    const kb = BigInt(k);
    let previous = 0n, current = 1n;
    for (let i = 0; i < n - 1; ++i) {
        const temp = current;
        current += previous * kb;
        previous = temp;
    }

    return current;
}

File: rust-rosalind/src/fib.rs
pub fn fib(n: u64, k: u64) -> u64 {
    let mut previous = 0;
    let mut current = 1;
    for _ in 0..n-1 {
        let temp = current;
        current += previous * k;
        previous = temp;
    }

    current
}